splunk_add_on_ucc_framework/generators/python_files/create_custom_command_python.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright 2025 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Any, Dict, Union
+import os
+from importlib import import_module
+import sys
+
+from splunk_add_on_ucc_framework.generators.python_files import PyGenerator
+
+GENERATED_FILES = {}
+
+
+class CustomCommandPy(PyGenerator):
+    __description__ = (
+        "Generates python files for custom commands provided in the globalConfig."
+    )
+
+    def _set_attributes(self, **kwargs: Any) -> None:
+        self.commands_info = []
+        if self._global_config and self._global_config.has_custom_search_commands():
+            for command in kwargs["custom_search_commands"]:
+                arguments = []
+                import_map = False
+                command["fileName"] = command["fileName"].replace(".py", "")
+                template = command["commandType"] + ".template"
+                if command["commandType"] == "reporting":
+                    src_pkg_bin = os.path.realpath(os.path.join(self._input_dir, "bin"))
+                    sys.path.insert(0, src_pkg_bin)
+                    if hasattr(import_module(command["fileName"]), "map"):
+                        import_map = True
+                    sys.path.pop(0)
```

Comment:
Can you please explain what are we trying to achieve here?

Reply:
For `reporting` type of custom search command, `map` method is optional, hence we check whether the developer has implemented this `map` method or not.